core/teca_algorithm_fwd.h

```diff
@@ -32,6 +32,7 @@ std::shared_ptr<T const> shared_from_this() const   \
 }
 
 #define TECA_ALGORITHM_CLASS_NAME(T)                \
+/** returns the string value "T" */                 \
```

Comment:
this is a virtual method and can be overriden in dervied classes. it does not necessarily return "T". say instead: returns the name of the class.